core/frontend/src/render/webgl/InstancedGeometry.ts

```diff
@@ -69,23 +71,105 @@ export interface PatternTransforms {
 }
 
 /** @internal */
-export class InstanceBuffers extends InstanceData {
-  private static readonly _patternParams = new Float32Array([0, 0, 0, 0]);
-
+export class InstanceBuffersData extends InstanceData {
   public readonly transforms: BufferHandle;
   public readonly featureIds?: BufferHandle;
-  public readonly hasFeatures: boolean;
   public readonly symbology?: BufferHandle;
-  public readonly patternParams = InstanceBuffers._patternParams;
-  public readonly patternTransforms = undefined;
-  public readonly viewIndependentOrigin = undefined;
+  private _noDispose = false;
 
-  private constructor(count: number, transforms: BufferHandle, rtcCenter: Point3d, range: Range3d, symbology?: BufferHandle, featureIds?: BufferHandle) {
-    super(count, rtcCenter, range);
+  private constructor(count: number, transforms: BufferHandle, rtcCenter: Point3d, symbology?: BufferHandle, featureIds?: BufferHandle, disableDisposal = false) {
+    super(count, rtcCenter);
     this.transforms = transforms;
     this.featureIds = featureIds;
-    this.hasFeatures = undefined !== featureIds;
     this.symbology = symbology;
+    this._noDispose = disableDisposal;
+  }
+
+  public static create(params: InstancedGraphicParams | InstancedGraphicProps, disableDisposal = false): InstanceBuffersData | undefined {
+    const { count, featureIds, symbologyOverrides, transforms } = params;
+
+    assert(count > 0 && Math.floor(count) === count);
+    assert(count === transforms.length / 12);
+    assert(undefined === featureIds || count === featureIds.length / 3);
+    assert(undefined === symbologyOverrides || count * 8 === symbologyOverrides.length);
+
+    let idBuf: BufferHandle | undefined;
+    if (undefined !== featureIds && undefined === (idBuf = BufferHandle.createArrayBuffer(featureIds)))
+      return undefined;
+
+    let symBuf: BufferHandle | undefined;
+    if (undefined !== symbologyOverrides && undefined === (symBuf = BufferHandle.createArrayBuffer(symbologyOverrides)))
+      return undefined;
+
+    const tfBuf = BufferHandle.createArrayBuffer(transforms);
+    const transformCenter = params.transformCenter instanceof Point3d ? params.transformCenter : Point3d.fromJSON(params.transformCenter);
+    if (!tfBuf) {
+      return undefined;
+    }
+
+    return undefined !== tfBuf ? new InstanceBuffersData(count, tfBuf, transformCenter, symBuf, idBuf, disableDisposal) : undefined;
```

Comment:
Aren't you already checking `tfBuf` right above this?

Reply:
Fixed.

Reply:
Please resolve comments when you are satisfied.